appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/UberJarCreator.java

```diff
@@ -241,6 +251,19 @@ public void buildUberJar() {
                     }
                 }
             }
+            
+            if (copyDirectory != null) {
+                String basePath = copyDirectory.getPath().replaceAll(copyDirectory + "$", "");
+                List<File> filesToCopy = fillFiles(copyDirectory);
+                for (File file : filesToCopy) {
+                    
+                        JarEntry deploymentEntry = new JarEntry(copyDirectory + file.getPath().replace(basePath, ""));
+                        jos.putNextEntry(deploymentEntry);
+                        Files.copy(file, jos);
```

Comment:
I think that it's not enough that JAR files are copied as is - due to how Payara Micro works since version 171, all loadable JAR files should be uncompressed. Is that right, @smillidge, @Pandrex247  ?
Otherwise an attempt to load compressed JARs (which is almost every 3rd party distributed JAR) would lead to ZIP exceptions like the one reported here: https://groups.google.com/d/msgid/payara-forum/0f0e28fb-0645-4992-bdf0-6201d09d3df0%40googlegroups.com